### ISSUE TITLE
Report

### ISSUE THREAD

**Issue by adramgo** [adrianrrrrr/ICM-ERA-ML#1]
Rank of importances. Most relevant variables may be lat, model_dir, lon,...  BUT collinearity wasn't inspected. 

![Figure_1](https://github.com/user-attachments/assets/cdff0fa8-ee2c-4dd0-820b-7b03fd68e88f)

Then,  we need to check the covariance matrix to perform a simple but relevant Exploratory Data Analysis (EDA). Gradient Boosting trees are, in principle, immune to collinearity. This makes they are also immune to overfitting. The train error will collapse, but test error will lower for bigger datasets. I am attaching an academic document part of the ETSETB-UPC Machine Learning, Master's Degree in Telecommunication's Engineering intensification in multimedia subject [Enric Moreno]. 

[Deliverable 12 Adrian.pdf](https://github.com/user-attachments/files/16347936/Deliverable.12.Adrian.pdf)

But it is not 100% true that boosting solves collinearity. CARTS with many number of features are trained sequentially by means of accomplishing the requirements set by a Loss Function as:

 $$\text{Loss}=\frac{1}{N}\sum_{i=1}^N (Y_i-\mathbf{Y}_i)^2$$

That in the attached work document, Adrian note as $\lambda$ regularization term. This gives a boosted tree algebraic representation for each sequential operation that should be executed in the CPU or the GPU exactly as:

$$F_{m+1}(x)=F_m(x)+\lambda\gamma h_m(x)$$

The results of comparing the XGBoost algorithm performed by the Evgeniia's Code and the XGBoost from Adrian's Code is in the following document:

[2020_smart_allvars_cpu_1500_test.xlsx](https://github.com/user-attachments/files/16359594/2020_smart_allvars_cpu_1500_test.xlsx)

Results:

<img src="https://github.com/user-attachments/assets/82938e3e-7759-4e52-93fc-c3ee7c8321c1" width="200" height="400" />

Can be stated that experiments were reproduced succesfully

<img width="968" alt="Screenshot 2024-07-18 at 17 14 20" src="https://github.com/user-attachments/assets/68bfeb2b-1e37-49e4-a00c-d06269fb77c7">
<img width="272" alt="Screenshot 2024-07-18 at 14 59 03" src="https://github.com/user-attachments/assets/3f2fe34a-d845-4444-8430-91ef049e1ff2">
